src/test/java/life/qbic/subscriptions/SubscriptionControllerTest.java

```diff
@@ -0,0 +1,132 @@
+  void encryptCancellationRequestDoesNotWorkForIncorrectInput() throws Exception {
+    var payload = new CancellationRequest("QABCD", "test@user.id");
+    var encrypted = "BStOJDfmn0ZyNceOPN3qU2xJw1mQfdbzY_a-uGt7Ae0=";
+    Mockito.when(requestEncrypter.encryptCancellationRequest(payload)).thenReturn(encrypted);
+
+    String json = String.format("{\"project\":\"%s\",\"user_id\":\"%s\"}", payload.project(), payload.userId());
+
+    mockMvc.perform(
+            get("/subscription/cancel")
+                .with(httpBasic("ChuckNorris","astrongpassphrase!"))
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(StandardCharsets.UTF_8)
+                .content(json)
+        )
+        .andExpect(status().is(400));
+  }
+
+  @Test
+  @DisplayName("decryptUnsubscriptionHash works")
+  void decryptUnsubscriptionHashWorks() throws Exception {
+    var payload = new CancellationRequest("QABCD", "test@user.id");
+    var encrypted = "BStOJDfmn0ZyNceOPN3qU2xJw1mQfdbzY_a-uGt7Ae0=";
+    Mockito.when(requestDecrypter.decryptCancellationRequest(encrypted)).thenReturn(payload);
+
+    String json = String.format("{\"project\":\"%s\",\"userId\":\"%s\"}", payload.project(), payload.userId());
+
+    mockMvc.perform(
+            post("/subscription/cancel/{encrypted}", encrypted)
+                .with(csrf())
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(StandardCharsets.UTF_8)
+        )
+        .andExpect(status().is(202))
+        .andExpect(content().string(json));
+  }
+
+  @Test
+  @DisplayName("decryptCancellationRequest does not work for incorrect input")
+  void decryptCancellationRequestDoesNotWorkForIncorrectInput() throws Exception {
+    var encrypted = "SomeInvalidToken";
+    Mockito.when(requestDecrypter.decryptCancellationRequest(encrypted))
+        .thenThrow(DecryptionException.class);
+
+    mockMvc
+        .perform(
+            post("/subscription/cancel/{encrypted}", encrypted)
+                .with(csrf())
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(StandardCharsets.UTF_8))
+        .andExpect(status().is(400));
+  }
+
+  @Test
+  @DisplayName("encryptCancellationRequest rejects unauthorized access")
+  void encryptCancellationRequestRejectsUnauthorizedAccess() throws Exception {
+    var payload = new CancellationRequest("QABCD", "test@user.id");
+    var encrypted = "BStOJDfmn0ZyNceOPN3qU2xJw1mQfdbzY_a-uGt7Ae0=";
+    Mockito.when(requestDecrypter.decryptCancellationRequest(encrypted)).thenReturn(payload);
+
+    String json = String.format("{\"project\":\"%s\",\"userId\":\"%s\"}", payload.project(), payload.userId());
+
+    mockMvc
+        .perform(
+            get("/subscription/cancel")
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding(StandardCharsets.UTF_8)
+                .content(json))
+        .andExpect(status().is(401));
+  }
+}
```

Comment:
```suggestion
}

```

Reply:
Why is this important?

Reply:
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
following the POSIX standard, L144 is not a line. That is also why Github thankfully highlights this for us.

Reply:
shared this also with the team now